src/components/App.js

```diff
@@ -18,17 +19,26 @@ export const App = ({ i18n, dir }) => {
   }, [dirState])
   // end of useDirection
 
+  // useFontSize
+  const [fontSizeClass, setFontSizeClass] = useState(articleTextSize.getFontSizeClassName())
+  useEffect(() => {
+    document.body.className = fontSizeClass
```

Comment:
To me, this is the only place where the concept of a "CSS class" is relevant. You could build it here and leave the rest of the code care about the logical values for font sizes (1...7) but not the class itself.